code/datums/components/sticky.dm

```diff
@@ -0,0 +1,102 @@
+// kinda like duct tape but not shit (just kidding, its shit but in a different way)
+/datum/component/sticky
+	var/atom/attached_to
+	var/icon/overlay
+
+/datum/component/sticky/Initialize(obj/item/I)
+	if(!isitem(parent))
+		stack_trace("/datum/component/sticky's parent is not an item, its [parent.type]")
+	RegisterSignal(parent, COMSIG_ITEM_PRE_ATTACK, PROC_REF(stick_to_it))
+
+/datum/component/sticky/proc/stick_to_it(obj/item/I, atom/target, mob/user, params)
+	SIGNAL_HANDLER
+	if(!in_range(I, target))
+		return
+	if(isstorage(target))
+		var/obj/item/storage/S = target
+		if(S.can_be_inserted(parent))
+			return
+	if(target.GetComponent(/datum/component/sticky))
+		return
```

Comment:
You can remove this in favor of making this component's `dupe_mode = COMPONENT_DUPE_UNIQUE`. The new component will be rejected and deleted if an object already has an identical one.

Reply:
I dont think that would do what this does. This check makes sure that the target isnt also sticky, so you cant put cameras on top of cameras

Reply:
Yeah you're right, resolving this